mcs/class/Mono.Profiler.Log/Mono.Profiler.Log/LogEventPool.cs

```diff
@@ -0,0 +1,36 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+
+namespace Mono.Profiler.Log {
+
+	sealed class LogEventPool {
```

Comment:
There are a few issues with this:

1. It won't work with non-seekable streams.
2. How do we sort all events between two sync points by timestamp in this design, while avoiding boxing?
3. You can't actually just read a random event in a log file because interpreting the values in that events requires having read all prior events in that buffer.